### ISSUE TITLE
Add support for 7z file format.

### ISSUE THREAD

**Issue by dc3-plaso** [log2timeline/dfvfs#23]

**Comment by joachimmetz** [comment]
Any suggestions for a Python module for this functionality?

**Comment by dc3-plaso** [comment]
I'm looking at the pylzma library, using their py7zlib module. This would open and decompress the files for us, however it will not allow for us to perform anytime of seeking into the files. Therefore, we will have to read in the whole files into a file stream in memory or temporary file.

**Comment by joachimmetz** [comment]
This one you mean: https://pypi.python.org/pypi/pylzma ?

For context what we normally look at regarding new dependencies: https://github.com/log2timeline/plaso/wiki/Adding-a-new-dependency

> Therefore, we will have to read in the whole files into a file stream in memory or temporary file.

This does not sound like an optimal solution, but we do somewhat the same for sqlite due to other limitations. I opt we get a temporary file manager in place before that or see if the seek functionality could be added to pylzma?

**Comment by dc3-plaso** [comment]
Agreed, I have currently implemented the 7z support using pylzma. It uses temporary files to read the files within the archive, but it should be trivial to change this to use the temporary file manager when its ready. Therefore, I would like to submit this for code review.

**Comment by joachimmetz** [comment]
python-lzma that is shipped with Ubuntu: https://launchpad.net/pyliblzma

**Comment by dc3-plaso** [comment]
I will be able to get back to writing this next week.

**Comment by joachimmetz** [comment]
Sounds good to me, I've been looking a xz compressed stream support. I might implement something based on Python lzma for that, seeing it's integrated in Python 3 and back ported for Python 2. However this will not provide 7z archive file support.
